File: SNPLogOCDemo/SNPLogOCDemo/SNPLog/SNPLogConfig.m
//
//  SNPLogConfig.m
//  SNPLogOCDemo
//
//  Created by zhengnan on 2026/9/3.
//

#import "SNPLogConfig.h"

@implementation SNPLogConfig

- (instancetype)initWithLogFilePath:(NSString *)logFilePath
                             deviceId:(NSString *)deviceId
                              logType:(SNPLogType)logType {
    if (self = [super init]) {
        _logFilePath = [logFilePath copy];
        _deviceId = [deviceId copy];
        _logType = logType;
    }
    return self;
}

- (instancetype)initWithLogFilePath:(NSString *)logFilePath
                             deviceId:(NSString *)deviceId {
    return [self initWithLogFilePath:logFilePath deviceId:deviceId logType:SNPLogTypeConsole];
}

@end

File: NanToolboxDemo/NanToolboxDemo/SNPLog/SNPLogConfig.swift
//
//  SNPLogConfig.swift
//  NanToolboxKit
//
//  Created by zhengnan on 2025/4/25.
//

import Foundation

public struct SNPLogConfig {
    public let logFilePath: String
    public let logType: SNPLogType
    public let deviceId: String
    
    public init(
        logFilePath: String,
        deviceId: String,
        logType: SNPLogType = .console
    ) {
        self.logFilePath = logFilePath
        self.deviceId = deviceId
        self.logType = logType
    }
}
